misc/dist/html/editor.html

```diff
@@ -254,9 +259,38 @@ <h2 id="welcome-modal-title">Important - Please read before continuing</h2>
 					>Web editor documentation</a> for usage instructions and limitations.
 				</p>
 			</div>
-			<button id="welcome-modal-dismiss" class="btn" type="button" onclick="closeWelcomeModal(true)" style="margin-top: 1rem">
-				OK, don't show again
-			</button>
+			<div id="welcome-modal-description-no-cross-origin-isolation" style="display: none">
+				<p>
+					The web server does not support cross-origin isolation,
+					which is required for the Godot Web Editor to function.
+				</p>
+				<p>
+					<strong>Reasons for cross-origin isolation being disabled:</strong>
+					<ul>
+						<li id="welcome-modal-reason-not-secure">
+							This page is not served from a secure context (HTTPS <i>or</i> localhost).
+						</li>
+						<li>
+							This page may not be served with cross-origin isolation headers
+							(check with the developer tools' Network tab).
+						</li>
```

Comment:
It's not possible to check whether the headers have been sent correctly – only whether cross-origin isolation is enabled (which itself depends on a secure context). The browser also does not inform us *why* cross-origin isolation is disabled, so we can only do guesswork here.